homeassistant/components/backup_sftp/backup.py

```diff
@@ -0,0 +1,155 @@
+            async with BackupAgentClient(self._entry, self._hass) as client:
+                return await client.iter_file(backup)
+        except (AssertionError, FileNotFoundError) as e:
+            raise BackupAgentError(
+                f"Unable to initiate download of backup id: {backup_id}. {e}"
+            ) from e
+        except Exception as e:
+            raise BackupAgentError(
+                f"Unexpected error while initiating download of backup id: {backup_id}. {e}"
+            ) from e
+
+    async def async_upload_backup(
+        self,
+        *,
+        open_stream: Callable[[], Coroutine[Any, Any, AsyncIterator[bytes]]],
+        backup: AgentBackup,
+        **kwargs: Any,
+    ) -> None:
+        """Upload a backup."""
+        LOGGER.debug("Received request to upload backup: %s", backup)
+        iterator = await open_stream()
+
+        try:
+            LOGGER.debug(
+                "Establishing SFTP connection to remote host in order to upload backup"
+            )
+            async with BackupAgentClient(self._entry, self._hass) as client:
+                LOGGER.debug("Uploading backup: %s", backup.backup_id)
+                await client.async_upload_backup(iterator, backup)
+        except Exception as e:
+            raise BackupAgentError(
+                f"Failed to upload backup to remote SFTP location. Error: {e}"
+            ) from e
+        LOGGER.debug("Successfully uploaded backup id: %s", backup.backup_id)
+
+    async def async_delete_backup(
+        self,
+        backup_id: str,
+        **kwargs: Any,
+    ) -> None:
+        """Delete a backup file from SFTP Backup Storage."""
+        LOGGER.debug("Received request to delete backup id: %s", backup_id)
+
+        try:
+            backup = await self.async_get_backup(backup_id)
+            LOGGER.debug(
+                "Establishing SFTP connection to remote host in order to delete backup"
+            )
+            async with BackupAgentClient(self._entry, self._hass) as client:
+                await client.async_delete_backup(backup)
+        except (AssertionError, SFTPError) as err:
+            raise BackupAgentError(
+                f"Failed to delete backup id: {backup_id}: {err}"
+            ) from err
+        except Exception as err:
+            raise BackupAgentError(
+                f"Unexpected error while removing backup: {backup_id}: {err}"
+            ) from err
+
+        LOGGER.debug("Successfully removed backup id: %s", backup_id)
+
+    async def async_list_backups(self, **kwargs: Any) -> list[AgentBackup]:
+        """List backups stored on SFTP Storage."""
+        try:
+            async with BackupAgentClient(self._entry, self._hass) as client:
+                return await client.async_list_backups()
+        except Exception as e:
+            LOGGER.exception(e)
+            LOGGER.error(
+                "Listing backups failed. Please review previous exception traceback"
+            )
```

Comment:
```suggestion
            LOGGER.exception("Listing backups failed.")
```
don't log both an error and a exception